numpy/random/mtrand/mtrand.pyx

```diff
@@ -4376,12 +4381,20 @@ cdef class RandomState:
         # not zero. We continue to use the SVD rather than Cholesky in
         # order to preserve current outputs. Note that symmetry has not
         # been checked.
+
         (u, s, v) = svd(cov)
-        neg = (np.sum(u.T * v, axis=1) < 0) & (s > 0)
+
+        neg = (np.sum(u.T * v, axis=1) < 0) & (s > 0) & ~np.isclose(s, 0,
+                                                                    atol=tol)
```

Comment:
doesn't the last part check now that we don't even have positive eigen values close to zero, i.e strictly positive definite?

didn't you want to add a symmetry check?